### PR TITLE
[TS] Hashes, now even better

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
The previous impl was actually only generating even values for equal row and column

**Implementation Description (bullets)**:
Without going into details, this should give us good enough has distribution while still being reasonably performant.

**Testing (What was existing testing like?  What have you done to improve it?)**:
Unit tests

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:
🔥

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/AbstractSweepQueueTest.java [hunk 1]
```diff
@@ -165,14 +166,12 @@ void putTimestampIntoTransactionTable(long ts, long commitTs) {
...
-        int counter = 0;
-        CellReference cellReference;
-        do {
-            byte[] colName = PtBytes.toBytes(counter);
-            cellReference = CellReference.of(tableRef, Cell.create(rowName, colName));
-            counter++;
-        } while (IntMath.mod(cellReference.hashCode(), shards) != 0);
-        return cellReference;
+
+        return IntStream.iterate(0, i -> i + 1)
```
Comment (berler) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/AbstractSweepQueueTest.java:170]: Is there any concern here that this implementation becomes more expensive as the number of shards increases?

Comment (berler) [reply atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/AbstractSweepQueueTest.java:170]: Just realized this seems to only be used for tests

File: atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CellReference.java [hunk 2]
```diff
@@ -15,31 +15,35 @@
...
+
     public abstract TableReference tableRef();
 
     public abstract Cell cell();
 
-    /**
-     * {@link Cell#hashCode()} implementation has a rather unfortunate case where it is always 0 if the row name and
-     * the column name match. We did not want to change it to keep backwards compatibility, but we need a uniform
-     * distribution here for all reasonable patterns.
-     */
```
Comment (jeremyk-91) [inline atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CellReference.java:31]: I think this context is still useful?

**Review by jeremyk-91 — APPROVED** [review]
🕐 0:03 🏁  🔥 

👍  removed MWR as I don't know if the comment is still useful, defer that to you

File: atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CellReference.java (hunk 2 above)
Comment (gmaretic) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CellReference.java:31]: makes sense

**Pull request merged**